src/Validation/ValidationSet.php

```diff
@@ -146,6 +146,10 @@ public function add(string $name, ValidationRule|array $rule)
         }
         $this->_rules[$name] = $rule;
 
+        if (null === $this->_allowEmpty) {
+            $this->_allowEmpty = false;
+        }
```

Comment:
```suggestion
        $this->_allowEmpty ??= false;
```

https://www.php.net/manual/en/migration74.new-features.php#migration74.new-features.core.null-coalescing-assignment-operator